src/sql/base/parts/editableDropdown/browser/dropdown.ts

```diff
@@ -267,14 +272,27 @@ export class Dropdown extends Disposable {
 
 	public set values(vals: string[] | undefined) {
 		if (vals) {
+			const longestString = vals.reduce((previous, current) => {
+				return previous.length > current.length ? previous : current;
+			}, '');
+
+			this._widthControlElement.innerText = longestString;
+
 			this._filter.filterString = '';
 			this._dataSource.options = vals.map(i => { return { value: i }; });
 			let height = this._dataSource.options.length * 22 > this._options.maxHeight! ? this._options.maxHeight! : this._dataSource.options.length * 22;
+
 			this._treeContainer.style.height = height + 'px';
-			this._treeContainer.style.width = DOM.getContentWidth(this._inputContainer) - 2 + 'px';
+
+			if (longestString.length > 10) {
+				this._treeContainer.style.width = this._widthControlElement.clientWidth + 'px';
+			}
+			this._treeContainer.style.maxWidth = `500px`;
```

Comment:
do you want to set a max width after the fact? Are you intending to limit it like this? i.e if there is a string with a length longer than this.

Reply:
Yeah, this is just basically a fail safe so it doesn't get infinitely large. It just made sense to have it here where I'm modifying the width - I can move it to where that treeContainer is being made instead though.